例外クラスの to_s が例外を発生させる実装になってるとバックトレースの表示が止まらなくなる? という不具合を修正しています。rb_ec_error_print() で同じ例外の出力を TAG_JUMP のせいで繰り返さないように変数でフラグを持って繰り返しの出力を抑制するようにしています。なるほど EC_PUSH_TAG() と EC_POP_TAG() の間で例外発生させると EC_EXEC_TAG() のところに戻ってしまうから無限ループ状態になってたのか。 [ruby-core:99172] [Bug #17033]
